src/index.js

```diff
@@ -176,29 +176,25 @@ class ReactTooltip extends React.Component {
 
   /* Look for the closest DOM root having tooltip and inject styles. */
   injectStyles() {
-    const { id } = this.props;
-    const targetArray = this.getTargetArray(id);
-    const domRoots = [];
-    targetArray.forEach(target => {
-      let parentNode = target.parentNode;
-      while (parentNode.parentNode && !parentNode.host) {
-        parentNode = parentNode.parentNode;
-      }
-      const head = parentNode.querySelector('head');
-      domRoots.push(head || parentNode);
-    });
-    if (domRoots.length) {
+    const { tooltipRef } = this;
+    if (!tooltipRef) {
+      return;
+    }
+
+    let parentNode = tooltipRef.parentNode;
+    while (parentNode.parentNode) {
+      parentNode = parentNode.parentNode;
+    }
+    const head = parentNode.querySelector('head');
+
+    const domRoot = parentNode || head;
+    // Prevent styles duplication.
+    if (!domRoot.querySelector('style[data-react-tooltip]')) {
       const style = document.createElement('style');
       style.textContent = baseCss;
```

Comment:
What about removing spaces & breaklines from output ?
`style.textContent = baseCss.replace(/(\r\n|\n|\r)/gm, '')).replace(/\s+/g, ' ')`

Reply:
Hey @louisgrasset

IMO css minify should be done in other PR as this PR only changes way of styles injection, neither styles content or form

Also I think css minify should be done via rollup or other tooling but not manually